Assets/Tile walking mini game/Scripts/GridManager.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GridManager : MonoBehaviour
+{
+    [SerializeField] private Tile[][] tileGrid;
+
+    [SerializeField] private TileColorManager tileColorManager;
+
+    [SerializeField] private ObstaclesPosition[][] obstacleLayouts;
+
+    public Action WinEvent;
+
+    public Action LoseEvent;
+    public int UntouchedTileCount { get; private set; }
+
+    private void Awake()
+    {
+        PopulateTileGrid();
+
+        CreateObstaclesLayouts();
+    }
+
+    private void Start()
+    {
+        ResetGrid();
+    }
+
+    public void ResetGrid()
+    {
+        foreach (Tile[] tileColumn in tileGrid)
+        {
+            foreach(Tile tile in tileColumn)

```

Comment:
Maybe set the obstacle layout first, then go through and set the ones that are obstacles to TileStatus.obstacle, and if they are not obstacle, set them to untouched? Should be able to cut down the number of loops

Reply:
I thought about this. I wasn't sure how to go about it. The problem I have is that, I still need to reset all the tiles to untouched before adding new obstacles. If I add first the obstacles when there are still obstacles and then set all the other tiles to untouched, this won't work because the previous obstacles won't be eliminated.